src/angularODataQuery.ts

```diff
@@ -258,4 +273,11 @@ export class ODataQuery<T> extends ODataOperation<T> {
     private extractArrayDataWithCount(res: HttpResponse<IODataResponseModel<T>>, config: ODataConfiguration): ODataPagedResult<T> {
         return config.extractQueryResultDataWithCount(res);
     }
+
+    private checkReservedCustomQueryOptionKey(key: string): string {
+        if (key.indexOf('$') === 0 || key.indexOf('@') === 0) {
```

Comment:
I think this method cannot handle undefined/null `key` ? Maybe add a null/undefined check?